### PR TITLE
fix(ui): do not display autologin shortcut when disabled

### LINKED ISSUE #7331: Disabling "Display Autologin shortcut" has no effect
Hi,

Even with "Display Autologin shortcut" disabled :
(in Administration / Centreon UI / Authentication properties)

set

Autologin link shows :

auto

Could we then correct this ?

Thank you 👍

Edit : issue still present in 19.10.0-beta.2.

### PR DESCRIPTION
**Pull request by adr-mo**
<h1> Pull Request Template </h1>

<h2> Description </h2>

As a *Centreon User*
When I *disable* the auto login shortcut in "Administration -> Parameters -> Centreon UI"
Then it is *not displayed* in the user menu from top counter

There is other cases to handle than the one that does not work currently.

Use cases to validate
* If the auto login is enabled, user has defined the auto login token and the option to hide the auto login key is *enabled* => The button is *NOT* displayed.
* If the auto login is enabled, user has defined the auto login token and the option to hide the auto login key is *disabled* => The button is  displayed.
* If the auto login is enabled, user has *NOT* defined the auto login token => The button is *NOT* displayed
* If the auto login is *disabled*, user has defined the auto login token and the option => The button is *NOT* displayed
* If the auto login is *disabled*, user has *NOT* defined the auto login token and the option => The button is *NOT* displayed

Fixes #7331 

<h2> Type of change </h2>

- [x] Patch fixing an issue (non-breaking change)

<h2> Target serie </h2>

- [x] 18.10.x
- [x] 19.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please check the description of the Pull Request

<h2> Checklist </h2>

<h5> Community contributors & Centreon team </h5>

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

<h5> Centreon team only </h5>

- [x] I have made sure that the **unit tests** related to the story are successful.
- [x] I have made sure that **unit tests covers 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/api/class/centreon_topcounter.class.php [hunk 1]
```diff
@@ -204,19 +208,53 @@ public function getUser()
...
         try {
             $res = $this->pearDB->query($query);
         } catch (\Exception $e) {
             throw new \RestInternalServerErrorException('Error getting the user.');
         }
 
-        if ($res->rowCount() === 0) {
-            throw new \RestUnauthorizedException('User does not exists.');
+        if ($row = $res->fetch()) {
+            $enableAutoLogin = $row['value'] == '1' ? true : false;
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]: As $enableAutoLogin and $enableAutoLoginShortcut are used only once, you could remove them.
And rename the 2 $row (lines 220 and 233) with something like $rowEnableShortcut and $rowEnableAutoLogin instead

File: www/api/class/centreon_topcounter.class.php [hunk 2]
```diff
@@ -204,19 +208,53 @@ public function getUser()
             $this->soundNotificationsEnabled = true;
         }
 
-        /* Get autologinkey */
-        $query = 'SELECT contact_autologin_key FROM contact WHERE contact_id = ' . (int)$user->user_id;
+        /* Is the autologin feature enabled ? */
+        $query = 'SELECT value FROM options WHERE options.key = "enable_autologin"';
 
         try {
             $res = $this->pearDB->query($query);
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
            $res = $this->pearDB->query(
                'SELECT value FROM options WHERE options.key = "enable_autologin"'
            );
```

Comment (sc979) [reply www/api/class/centreon_topcounter.class.php]: this way, you doen't need to declare another variable on line 212 ;)

**Comment by adr-mo** [comment]
I've made the changes requested and rebased the branch @sc979

File: www/api/class/centreon_topcounter.class.php [hunk 3]
```diff
@@ -204,19 +206,49 @@ public function getUser()
...
+        $rowEnableAutoLogin = $res->fetch();
+
+        /* If the autologin feature is enabled then fetch the autologin key
+         * And display the shortcut if the option is enabled
+         */
+        if ($rowEnableAutoLogin['value'] == '1' && $rowEnableShortcut['value'] == '1') {
+            /* Get autologinkey */
+            try {
+                $res = $this->pearDB->query(
+                    $query = 'SELECT contact_autologin_key FROM contact WHERE contact_id = ' . (int)$user->user_id;
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
                    'SELECT contact_autologin_key FROM contact WHERE contact_id = :userID'
```

File: www/api/class/centreon_topcounter.class.php [hunk 4]
```diff
@@ -204,19 +206,49 @@ public function getUser()
...
-            throw new \RestUnauthorizedException('User does not exists.');
+        $rowEnableAutoLogin = $res->fetch();
+
+        /* If the autologin feature is enabled then fetch the autologin key
+         * And display the shortcut if the option is enabled
+         */
+        if ($rowEnableAutoLogin['value'] == '1' && $rowEnableShortcut['value'] == '1') {
+            /* Get autologinkey */
+            try {
+                $res = $this->pearDB->query(
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
                $res = $this->pearDB->prepare(
```

File: www/api/class/centreon_topcounter.class.php [hunk 5]
```diff
@@ -204,19 +206,49 @@ public function getUser()
...
+                    $query = 'SELECT contact_autologin_key FROM contact WHERE contact_id = ' . (int)$user->user_id;
+                );
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
                );
                $res->bindValue(':userId',  (int)$user->user_id, \PDO::PARAM_INT);
                $res->execute();
```

File: www/api/class/centreon_topcounter.class.php [hunk 6]
```diff
@@ -204,19 +206,49 @@ public function getUser()
...
+            } catch (\Exception $e) {
+                throw new \RestInternalServerErrorException('Error getting the user.');
+            }
+
+            if ($res->rowCount() === 0) {
+                throw new \RestUnauthorizedException('User does not exists.');
+            }
+
+            if ($row = $res->fetch()) {
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
            $row = $res->fetch())
```

File: www/api/class/centreon_topcounter.class.php [hunk 7]
```diff
@@ -204,19 +206,49 @@ public function getUser()
...
+                $autoLoginKey = $row['contact_autologin_key'];
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
            $autoLoginKey = $row['contact_autologin_key'];
```

File: www/api/class/centreon_topcounter.class.php [hunk 8]
```diff
@@ -204,19 +206,49 @@ public function getUser()
...
         }
 
-        if ($res->rowCount() === 0) {
-            throw new \RestUnauthorizedException('User does not exists.');
+        $rowEnableAutoLogin = $res->fetch();
+
+        /* If the autologin feature is enabled then fetch the autologin key
+         * And display the shortcut if the option is enabled
+         */
+        if ($rowEnableAutoLogin['value'] == '1' && $rowEnableShortcut['value'] == '1') {
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
        if (isset($rowEnableAutoLogin['value'])
            && isset($rowEnableShortcut['value'])
            && $rowEnableAutoLogin['value'] === '1'
            && $rowEnableShortcut['value'] === '1'
        ) {
```

File: www/api/class/centreon_topcounter.class.php [hunk 9]
```diff
@@ -204,19 +206,49 @@ public function getUser()
...
+            $res = $this->pearDB->query(
+                'SELECT value FROM options WHERE options.key = "enable_autologin"';
+            )
+        } catch (\Exception $e) {
+            throw new \RestInternalServerErrorException('Error getting the user.');
+        }
 
+        $rowEnableShortcut = $res->fetch();
+
+	    /* Do we need to display the autologin shortcut ? */
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
	 /* Do we need to display the autologin shortcut ? */
```

File: www/api/class/centreon_topcounter.class.php (hunk 6 above)
Comment (adr-mo) [reply www/api/class/centreon_topcounter.class.php]: done

File: www/api/class/centreon_topcounter.class.php (hunk 5 above)
Comment (adr-mo) [reply www/api/class/centreon_topcounter.class.php]: done

File: www/api/class/centreon_topcounter.class.php (hunk 4 above)
Comment (adr-mo) [reply www/api/class/centreon_topcounter.class.php]: done

File: www/api/class/centreon_topcounter.class.php (hunk 3 above)
Comment (adr-mo) [reply www/api/class/centreon_topcounter.class.php]: done

File: www/api/class/centreon_topcounter.class.php [hunk 10]
```diff
@@ -204,19 +206,54 @@ public function getUser()
...
+            } catch (\Exception $e) {
+                throw new \RestInternalServerErrorException('Error getting the user.');
+            }
+
+            if ($res->rowCount() === 0) {
+                throw new \RestUnauthorizedException('User does not exists.');
+            }
+
+            $row = $res->fetch();
+            $autoLoginKey = $row['contact_autologin_key'];
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
            $autoLoginKey = $row['contact_autologin_key'] ?? null;
```

File: www/api/class/centreon_topcounter.class.php [hunk 11]
```diff
@@ -191,6 +191,8 @@ public function putAutoLoginToken()
      */
     public function getUser()
     {
+        $autoLoginKey = null;
```
Comment (sc979) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
```

File: www/api/class/centreon_topcounter.class.php [hunk 12]
```diff
@@ -204,19 +204,54 @@ public function getUser()
...
+        $rowEnableAutoLogin = $res->fetch();
+
+        /* If the autologin feature is enabled then fetch the autologin key
+         * And display the shortcut if the option is enabled
+         */
+        if (isset($rowEnableAutoLogin['value'])
+            && isset($rowEnableShortcut['value'])
+            && $rowEnableAutoLogin['value'] === '1'
+            && $rowEnableShortcut['value'] === '1'
+            ) {
```
Comment (kduret) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
        ) {
```

File: www/api/class/centreon_topcounter.class.php [hunk 13]
```diff
@@ -204,19 +204,54 @@ public function getUser()
...
+                    'SELECT contact_autologin_key FROM contact WHERE contact_id = :userID'
+                );
+                $res->bindValue(':userId',  (int)$user->user_id, \PDO::PARAM_INT);
+                $res->execute();
+            } catch (\Exception $e) {
+                throw new \RestInternalServerErrorException('Error getting the user.');
+            }
+
+            if ($res->rowCount() === 0) {
+                throw new \RestUnauthorizedException('User does not exists.');
```
Comment (kduret) [inline www/api/class/centreon_topcounter.class.php]:
```suggestion
                throw new \RestUnauthorizedException('User does not exist.');
```

File: www/api/class/centreon_topcounter.class.php (hunk 12 above)
Comment (adr-mo) [reply www/api/class/centreon_topcounter.class.php]: done

File: www/api/class/centreon_topcounter.class.php (hunk 13 above)
Comment (adr-mo) [reply www/api/class/centreon_topcounter.class.php]: done

**Comment by CPbN** [comment]
Still facing this small issue in 19.10.0-beta.2.
Perhaps we could then merge this tiny fix.
Thank you 👍

**Pull request merged**